### PR TITLE
Fixes #10356: Add a "test" option to ncf-setup

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/10356

### CODE REVIEW

**Comment by amousset** [comment]
Commit modified

File: scripts/rudder-setup/setup-ncf.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ setup_ncf() {
     # cfengine from Rudder
     RUDDER_VERSION=$(echo "${CFENGINE_VERSION}" | cut -f 2 -d "-")
     add_repo
-    setup_agent
+    setup_agent || true
```
Comment (peckpeck) [inline scripts/rudder-setup/setup-ncf.sh]: Why ?

File: scripts/rudder-setup/test-ncf.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    ${PM_INSTALL} ${TEST_DEPENDENCIES}
+
+    # Start the test suite
+    echo "Beginning tests, using CFEngine version \"`cf-agent -V`\""
+
+    # Use path from Rudder package if present
+    export PATH="${PATH}:/var/rudder/cfengine-community/bin/"
+
+    make test-unsafe
+}
```
Comment (peckpeck) [inline scripts/rudder-setup/test-ncf.sh]: Missing newline

File: scripts/rudder-setup/test-ncf.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    # Install dependencies
+    ${PM_INSTALL} ${TEST_DEPENDENCIES}
+
+    # Start the test suite
+    echo "Beginning tests, using CFEngine version \"`cf-agent -V`\""
+
+    # Use path from Rudder package if present
+    export PATH="${PATH}:/var/rudder/cfengine-community/bin/"
+
+    make test-unsafe
```
Comment (peckpeck) [inline scripts/rudder-setup/test-ncf.sh]: Shouldn't this be a parameter ?

File: scripts/rudder-setup/test-ncf.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    cd "${directory}"
+
+    # Install dependencies
+    ${PM_INSTALL} ${TEST_DEPENDENCIES}
+
+    # Start the test suite
+    echo "Beginning tests, using CFEngine version \"`cf-agent -V`\""
+
+    # Use path from Rudder package if present
+    export PATH="${PATH}:/var/rudder/cfengine-community/bin/"
```
Comment (peckpeck) [inline scripts/rudder-setup/test-ncf.sh]: This should probably be before the call to cf-agent

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Pull request merged**